numpy/testing/tests/test_utils.py

```diff
@@ -468,48 +468,117 @@ def test_report_fail_percentage(self):
         self.assertTrue("mismatch 25.0%" in msg)
 
 class TestArrayAlmostEqualNulp(unittest.TestCase):
-    @dec.knownfailureif(True, "Github issue #347")
+    def setUp(self):
+        self.seed = 12345
+
+    def _failure(self, x, y, nulp):
+        assert_array_almost_equal_nulp(x, y, nulp)
+
     def test_simple(self):
-        np.random.seed(12345)
-        for i in range(100):
-            dev = np.random.randn(10)
-            x = np.ones(10)
-            y = x + dev * np.finfo(np.float64).eps
-            assert_array_almost_equal_nulp(x, y, nulp=2 * np.max(dev))
+        r = np.random.RandomState(self.seed)
+        dev = r.randn(10)
+        dev = dev/np.max(np.abs(dev))
+        x = np.ones(10)
+        y = x + dev * np.finfo(np.float64).eps
+        assert_array_almost_equal_nulp(x, y, nulp=1)
 
     def test_simple2(self):
-        x = np.random.randn(10)
+        r = np.random.RandomState(self.seed)
+        x = r.randn(10)
         y = 2 * x
-        def failure():
-            return assert_array_almost_equal_nulp(x, y,
-                                                  nulp=1000)
-        self.assertRaises(AssertionError, failure)
+        def failure(x, y):
```

Comment:
The function definition would look better with blank lines before and after. Although I think you could dispense with the function and use `assert_array_almost_equal_nulp` directly.